service/application/src/main/java/com/solace/maas/ep/event/management/agent/subscriber/messageProcessors/ScanCommandMessageProcessor.java

```diff
@@ -68,7 +73,33 @@ public void processMessage(ScanCommandMessage message) {
 
         log.info("Received scan request {}. Request details: {}", scanRequestBO.getScanId(), scanRequestBO);
 
-        scanManager.scan(scanRequestBO);
+        String scanId = scanManager.scan(scanRequestBO);
+        waitForScanCompletion(scanId);
+        //message will be acked
+    }
+
+    public void waitForScanCompletion(String scanId) {
+        try {
+            await()
+                    .atMost(5, MINUTES)
+                    .pollInterval(10, SECONDS)
+                    .pollInSameThread()
+                    .until(() -> {
+                        try {
+                            return waitUntilScanIsCompleted(scanId);
+                        } catch (Exception e) {
+                            log.error("Error while waiting for scan to complete", e);
+                            return false;
+                        }
+                    });
+        } catch (ConditionTimeoutException e) {
+            // Handle the timeout scenario as needed
+            log.error("Scan with id {} did not complete within the expected time", scanId);
+        }
+    }
```

Comment:
IMO, we should only do this if CEMA is running on managed mode. It is very easy to do so I will get back to this